packages/react/src/runtime/style-cache.tsx

```diff
@@ -12,13 +12,43 @@ const Cache: any = isNodeEnvironment() ? createContext<Record<string, true> | nu
 if (!isNodeEnvironment()) {
   /**
    * Iterates through all found style elements generated when server side rendering.
-   *
-   * @param cb
    */
   const ssrStyles = document.querySelectorAll<HTMLStyleElement>('style[data-cmpld]');
   for (let i = 0; i < ssrStyles.length; i++) {
+    // Create the client cache for all SSR'd classes
+    const stylesheet = ssrStyles[i];
+    const rulesText = stylesheet.innerText;
+
+    if (rulesText) {
+      // We split the stylesheet innerText by the closing } then loop over the array
+      const arr = rulesText.split('}');
+
+      // Nested rule string
+      let nestedRule = '';
+
+      for (let i = 0; i < arr.length; i++) {
+        const myRule = `${arr[i]}}`;
+        // check if rule starts with @
+        if (myRule.charCodeAt(0) === 64) {
+          // begin building nested rule string
+          nestedRule = myRule;
+          // if the nested rule ends with }} then it's considered complete
+        } else if (nestedRule.substr(-2) === '}}') {
+          // add nested rule to Cache and reset nestedRule string
+          Cache[nestedRule] = true;
+          nestedRule = '';
+          // if we have started building a nested rule but it is not complete then append an }
+        } else if (nestedRule) {
+          nestedRule += '}';
+        } else {
+          // if not an @ rule then just add to Cache
+          Cache[myRule] = true;
+        }
+      }
+    }
+
     // Move all found server-side rendered style elements to the head before React hydration happens.
-    document.head.appendChild(ssrStyles[i]);
+    document.head.appendChild(stylesheet);
```

Comment:
hmm there's a high chance the styles we add won't be ordered, which could create some problems. as a fix we can use `insertRule`:

```
import insertRule from './sheet';

insertRule(myRule);
```